craft_parts/sources/base.py

```diff
@@ -178,7 +178,6 @@ def __init__(
 
     # pylint: enable=too-many-arguments
 
-    @abc.abstractmethod
```

Comment:
from my understanding of the PR, it looks like you removed this `abstractmethod` decorator here because `FileSource` doesn't need to do any provisioning, right? If that's the case, I think it might be preferable to keep the `abstractmethod` and have `FileSource` override it with an empty implementation. It's not a strong preference but I believe it's good that future `FileSourceHandler` specializations *have* to implement this, even if as a no-op, because it means they will at least have to think about.

Reply:
The overall structure of the sources should be refactored at some point to properly handle vcs-related sources (that accept `source-tag`, `source-branch`, `source-commit`, etc) but until then I'll do as suggested and keep the abstract method, overriding it in the plain file source handler implementation.